cmd/loadgenerator/main.go

```diff
@@ -177,12 +181,12 @@ func generateTransactions(restClient client.RestClient, cfg config, privateKey *
 		go func(base int) {
 			defer sendWaitGroup.Done()
 			for x := base; x < transactionBlockSize; x += nroutines {
-				_, err = restClient.SendRawTransaction(txns[x])
-				if err != nil {
-					if strings.Contains(err.Error(), "txn dead") || strings.Contains(err.Error(), "below threshold") {
+				_, err2 := restClient.SendRawTransaction(txns[x])
+				if err2 != nil {
+					if strings.Contains(err2.Error(), "txn dead") || strings.Contains(err2.Error(), "below threshold") {
 						break
 					}
-					fmt.Fprintf(os.Stderr, "unable to send transaction : %v\n", err)
+					fmt.Fprintf(os.Stderr, "unable to send transaction : %v\n", err2)
```

Comment:
the function doesn't return the `err` variable, so I can't see why you would need to allocate another `err2` variable.. anything that I'm missing ?

Reply:
I was seeing a memory segmentation fault when this is handling errors.

Reply:
huh.. I think that you're correct in that it should be a unique variable per go-routine.
Let me confirm that tomorrow morning with race enabled.